pkg/datastore/mysql/query.go

```diff
@@ -93,3 +98,47 @@ func toMySQLDirection(d datastore.OrderDirection) string {
 		return ""
 	}
 }
+
+func refineOrdersField(orders []datastore.Order) []datastore.Order {
+	for i, order := range orders {
+		switch order.Field {
+		case "SyncState.Status":
+			orders[i].Field = "SyncState"
+		default:
+			continue
+		}
+	}
+	return orders
+}
+
+func refineFiltersField(filters []datastore.ListFilter) []datastore.ListFilter {
+	for i, filter := range filters {
+		switch filter.Field {
+		case "SyncState.Status":
+			filters[i].Field = "SyncState"
+		default:
+			continue
+		}
+	}
+	return filters
+}
+
+func refineFiltersOperator(filters []datastore.ListFilter) ([]datastore.ListFilter, error) {
+	var err error
+	for i, filter := range filters {
+		switch filter.Operator {
+		case "==":
+			filters[i].Operator = "="
+		case "in":
+			filters[i].Operator = "IN"
+		case "not-in":
+			filters[i].Operator = "NOT IN"
+		case "!=", ">", ">=", "<", "<=":
+			continue
+		default:
+			err = fmt.Errorf("unsupported operator %s", filter.Operator)
```

Comment:
Why not returning here?

Reply:
You're right 👍 We should just return err from here 👍